.github/workflows/main.yml

```diff
@@ -56,22 +56,23 @@ jobs:
         steps:
             - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
 
-            - name: Init Python
-              uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5
+            - name: Install uv
+              uses: astral-sh/setup-uv@v4
               with:
+                  version: "0.5.10"
                   python-version: "3.11"
-                  cache: "pip"
+                  enable-cache: true
+
 
             - name: Install Python dependencies
               run: |
-                  python -m pip install --upgrade pip
-                  pip install -e pytest-pixl/
-                  pip install -e pixl_core/[test]
-                  pip install -e ${{ matrix.package_dir }}/[test]
+                  uv sync --all-extras --all-packages
```

Comment:
to avoid needing the `--all-extras` and `--all-packages` flags, you can add a `dev` [dependency group](https://docs.astral.sh/uv/concepts/projects/dependencies/#development-dependencies) in the top-level `pyproject.toml`(you can move dev dependencies from all packages into the top-level toml). The dev dependencies are installed by default when you do `uv sync`.

Also, it's probably best to do `uv sync --frozen` to avoid updating the `uv.lock` file before running tests

Reply:
Should have fixed this now. It looked like we weren't making a huge distinction between dev and test optional packages, so I merged them all into a top-level `dev` dependency group.

Re the `--frozen` option, since we now have the pre-commit check making sure that the lock file has been updated, I think I'm going to run the tests without this option, so that in the event that you forget to update the lock file before pushing, you get the test results for the lock file you were *supposed* to have, which is I think more useful than using the old version. Unless you object.

Reply:
> I think I'm going to run the tests without this option, so that in the event that you forget to update the lock file before pushing, you get the test results for the lock file you were supposed to have

yeah I think you're right actually, that makes more sense